app/actions/memberActions.js

```diff
@@ -16,28 +20,148 @@ export function memberInfoIsLoading(bool) {
   }
 }
 
-export function memberInfoFetchDataSuccess(memberInfo) {
-  return {
-    type: MEMBER_INFO_FETCH_DATA_SUCCESS,
-    memberInfo
+export function handleMemberInfoResponse(response) {
+  switch (response.status) {
+    case 200:
+      return {
+        type: MEMBER_INFO_FETCH_DATA_SUCCESS,
+        memberInfo: response.data
+      }
+
+    default:
+      return {
+        type: MEMBER_INFO_HAS_ERRORED,
+        errorMessage: response.data.message
+      }
   }
 }
 
 export function memberInfoFetchData() {
   return dispatch => {
     dispatch(memberInfoIsLoading(true))
-    apiFetch('/api/ping')
+    axios({ url: '/api/ping', withCredentials: true })
       .then(response => {
-        if (!response.ok) {
-          throw Error(response.statusText)
+        if (response) {
+          dispatch(handleMemberInfoResponse(response))
+          dispatch(memberInfoIsLoading(false))
         }
+      })
+      .catch(error => {
+        if (error.response) {
+          dispatch(handleMemberInfoResponse(error.response))
+          dispatch(memberInfoIsLoading(false))
+        }
+      })
+  }
+}
 
-        dispatch(memberInfoIsLoading(false))
+// LOAD USER REGISTRATION DETAILS FOR SIGNUP COMPLETION FORM
+export function loadCompleteSignupLoading(bool) {
+  return {
+    type: LOAD_COMPLETE_SIGNUP_LOADING,
+    loadCompleteSignupLoading: bool
+  }
+}
 
-        return response
+export function handleLoadCompleteSignupResponse(response) {
+  switch (response.status) {
+    case 200:
+      return {
+        type: LOAD_COMPLETE_SIGNUP_SUCCESS,
+        data: response.data
+      }
+
+    default:
+      return {
+        type: LOAD_COMPLETE_SIGNUP_FAILURE,
+        errorMessage: response.data.message
+      }
+  }
+}
+
+export function loadCompleteSignup(token) {
+  return dispatch => {
+    dispatch(loadCompleteSignupLoading(true))
+    axios({ url: `/api/signup/validate-invite/${token}`, withCredentials: true })
+      .then(response => {
+        if (response) {
+          dispatch(loadCompleteSignupLoading(false))
+          dispatch(handleLoadCompleteSignupResponse(response))
+        }
+      })
+      .catch(error => {
+        if (error.response) {
+          dispatch(handleLoadCompleteSignupResponse())
+          dispatch(loadCompleteSignupLoading(false))
+        }
+      })
+  }
+}
+
+// PROVIDE PASSWORD AND/OR NAME TO COMPLETE ACCOUNT CREATION
+export function createUserInitiate(bool) {
+  return {
+    type: CREATE_USER_LOADING,
+    createUserLoading: bool
+  }
+}
+
+export function handleCreateUserResponse(response) {
+  switch (response.status) {
+    case 200:
+      return {
+        type: CREATE_USER_SUCCESS,
+        data: response.data
+      }
+
+    case 409:
+      return {
+        type: CREATE_USER_DUPLICATE_FAILURE,
+        errorMessage: response.data.message
+      }
+
+    case 400:
```

Comment:
The auth/view create user explicitly returns 200, 400, 409 and 500 but yes there is little value in handling 400 and 500 instead of having them fall through to default